woocommerce/woocommerce
# Also read from posts when reading from COT to check for direct write

### All Submissions:






### Changes proposed in this Pull Request:

If sync is enabled, then while reading an order from custom order tables, we also read from posts table and then compare the two records. If there is a discrepancy between the post record and order record, then we will update the order record with the changes from the post record (the assumption is that post record had direct write, which led to the discrepancy).



### How to test the changes in this Pull Request:

Check the unit tests and see if you think they are adequate. Additionally:

1. Create a new order (or edit an existing one) while having custom order tables as authoritative and sync enabled.
2. Add/update/delete a post meta record manually for that order.
3. Refresh the order edit page so that order is read again. Post meta changes should have been transferred to the order as well.

### Other information:




### FOR PR REVIEWER ONLY:
